### PR TITLE
GH-35369: [Docs] Fix reST markup around ref:`IPC format <format-ipc>`

### LINKED ISSUE #35369: [Docs] reStructuredText link markup is broken
### Describe the bug, including details regarding any error messages, version, and platform.

https://arrow.apache.org/docs/format/Glossary.html#term-record-batch

> record batch
>
> In the :ref:\`IPC format \<format-ipc\>\`: the primitive unit of data. A record batch consists of an ordered list of [buffers](https://arrow.apache.org/docs/format/Glossary.html#term-buffer) corresponding to a [schema](https://arrow.apache.org/docs/format/Glossary.html#term-schema).

":ref:\`IPC format \<format-ipc\>\`" should be linked to https://arrow.apache.org/docs/dev/format/Columnar.html#format-ipc .

### Component(s)

Documentation

### PR DESCRIPTION
**Pull request by Divyansh200102**
* Closes: #35369

### CODE REVIEW

**Comment by kou** [comment]
Could you read the auto-generated comment?
https://github.com/apache/arrow/pull/38276#issuecomment-1763484822

And could you read our pull request template and follow it instead of remove it entirely without reading it?

File: docs/source/cpp/tables.rst [hunk 1]
```diff
@@ -82,7 +82,7 @@ and computation functions, possibly incremental.
          structure as described in text above.
 
 Record batches can be sent between implementations, such as via 
-:ref:`IPC <format-ipc>` or
+:ref:`IPC format <format-ipc>` or
```
Comment (kou) [inline docs/source/cpp/tables.rst:85]: Is this really needed?
https://arrow.apache.org/docs/cpp/tables.html#record-batches renders correctly without this:

> Record batches can be sent between implementations, such as via [IPC](https://arrow.apache.org/docs/format/Columnar.html#format-ipc) or via ...

File: docs/source/developers/overview.rst [hunk 2]
```diff
@@ -184,7 +184,7 @@ implementation can support:
    same endianness).  This includes ancillary functionality such as reading
    and writing various file formats, such as Parquet.
 2. Cross endian support (implementations will do byte reordering when
-   appropriate for :ref:`IPC <format-ipc>` and :ref:`Flight <flight-rpc>`
+   appropriate for :ref:`IPC format <format-ipc>` and :ref:`Flight <flight-rpc>`
```
Comment (kou) [inline docs/source/developers/overview.rst:187]: ditto.

https://arrow.apache.org/docs/developers/overview.html#endianness

> (implementations will do byte reordering when appropriate for [IPC](https://arrow.apache.org/docs/format/Columnar.html#format-ipc) and...

File: docs/source/java/vector_schema_root.rst [hunk 3]
```diff
@@ -107,7 +107,7 @@ Here is an example of creating a VectorSchemaRoot:
 
 Data can be loaded into/unloaded from a VectorSchemaRoot via `VectorLoader`_
 and `VectorUnloader`_.  They handle converting between VectorSchemaRoot and
-`ArrowRecordBatch`_ (a representation of a RecordBatch :ref:`IPC <format-ipc>`
+`ArrowRecordBatch`_ (a representation of a RecordBatch :ref:`IPC format <format-ipc>`
```
Comment (kou) [inline docs/source/java/vector_schema_root.rst:110]: ditto.

https://arrow.apache.org/docs/java/vector_schema_root.html#vectorschemaroot

> (a representation of a RecordBatch [IPC](https://arrow.apache.org/docs/format/Columnar.html#format-ipc) message)

**Review by kou — COMMENTED** [review]
Could you check your change on local?
See also: https://arrow.apache.org/docs/developers/documentation.html#building-with-docker

I think that the following is the needed change:

```diff
diff --git a/docs/source/format/Glossary.rst b/docs/source/format/Glossary.rst
index 65d6e0afa..d0adc0cc3 100644
--- a/docs/source/format/Glossary.rst
+++ b/docs/source/format/Glossary.rst
@@ -174,7 +174,7 @@ Glossary
        .. seealso:: :term:`data type`
 
    record batch
-       **In the :ref:`IPC format <format-ipc>`**: the primitive unit
+       **In the :ref:`IPC format <format-ipc>` **: the primitive unit
        of data.  A record batch consists of an ordered list of
        :term:`buffers <buffer>` corresponding to a :term:`schema`.
 
```

**Comment by Divyansh200102** [comment]
> Could you check your change on local? See also: https://arrow.apache.org/docs/developers/documentation.html#building-with-docker
> 
> I think that the following is the needed change:
> 
> ```diff
> diff --git a/docs/source/format/Glossary.rst b/docs/source/format/Glossary.rst
> index 65d6e0afa..d0adc0cc3 100644
> --- a/docs/source/format/Glossary.rst
> +++ b/docs/source/format/Glossary.rst
> @@ -174,7 +174,7 @@ Glossary
>         .. seealso:: :term:`data type`
>  
>     record batch
> -       **In the :ref:`IPC format <format-ipc>`**: the primitive unit
> +       **In the :ref:`IPC format <format-ipc>` **: the primitive unit
>         of data.  A record batch consists of an ordered list of
>         :term:`buffers <buffer>` corresponding to a :term:`schema`.
>  
> ```

hi, @kou  i am trying to setup the project using the docker-compose.yml file by using the command "docker compose up -d" but this is error is showing up (Error response from daemon: manifest for amd64/maven:3.5.4-eclipse-temurin-8 not found: manifest unknown: manifest unknown) i think this version of amd64/maven is not in the registry .i tried pulling manually as well but still nothing can you help me out to solve this?

**Comment by raulcd** [comment]
Hi @Divyansh200102 we use docker-compose as a utility to help us run different testing containers mainly for CI and some locally but we don't use `docker-compose up` as we have lots of images and this is not supposed to work.
Due to the large amount of languages and tools in the Apache Arrow repository we have in this contribution guide several examples on how to build and test locally for the different languages. Please take a look there and don't hesitate on asking if something is not clear enough: https://arrow.apache.org/docs/dev/developers/guide/index.html

**Comment by raulcd** [comment]
BTW if you are trying to build the documentation locally here is the documentation: https://arrow.apache.org/docs/developers/documentation.html

**Comment by Divyansh200102** [comment]
@raulcd 
i am encountering this
image

**Comment by Divyansh200102** [comment]
[](<script src="https://gist.github.com/Divyansh200102/7e26fc47fd2f66a438970c5d8f1fae4f.js"></script>)

**Comment by Divyansh200102** [comment]
@raulcd i am encountering this
https://gist.github.com/Divyansh200102/7e26fc47fd2f66a438970c5d8f1fae4f

**Comment by Divyansh200102** [comment]
@raulcd i am encountering this
https://gist.github.com/Divyansh200102/7e26fc47fd2f66a438970c5d8f1fae4f

**Comment by Divyansh200102** [comment]
i am encountering this @raulcd 
https://gist.github.com/Divyansh200102/7e26fc47fd2f66a438970c5d8f1fae4f

**Comment by raulcd** [comment]
Hi @Divyansh200102 , it seems to be failing on building pygit2 which is a dependency for Archery:
```
Failed to build pygit2
ERROR: Could not build wheels for pygit2, which is required to install pyproject.toml-based projects
```
I am not entirely sure what the problem is with the logs provided but you are probably using Python 3.12 and that's why there are no wheels available yet for them as seen on PyPI: https://pypi.org/project/pygit2/#files
If you want to build from source you'll need to follow their installation guide requirements for Windows: https://www.pygit2.org/install.html#installing-on-windows

**Comment by Divyansh200102** [comment]
> Hi @Divyansh200102 , it seems to be failing on building pygit2 which is a dependency for Archery:
> 
> ```
> Failed to build pygit2
> ERROR: Could not build wheels for pygit2, which is required to install pyproject.toml-based projects
> ```
> 
> I am not entirely sure what the problem is with the logs provided but you are probably using Python 3.12 and that's why there are no wheels available yet for them as seen on PyPI: https://pypi.org/project/pygit2/#files If you want to build from source you'll need to follow their installation guide requirements for Windows: https://www.pygit2.org/install.html#installing-on-windows

**Comment by Divyansh200102** [comment]
@raulcd please see this https://gist.github.com/Divyansh200102/3ba4f5e391d8e62307f8b584a5a659d8

**Comment by Divyansh200102** [comment]
Please review the changes @kou

File: docs/source/java/vector_schema_root.rst [hunk 4]
```diff
@@ -107,7 +107,7 @@ Here is an example of creating a VectorSchemaRoot:
 
 Data can be loaded into/unloaded from a VectorSchemaRoot via `VectorLoader`_
 and `VectorUnloader`_.  They handle converting between VectorSchemaRoot and
-`ArrowRecordBatch`_ (a representation of a RecordBatch :ref:`IPC <format-ipc>`
+`ArrowRecordBatch`_ (a representation of a RecordBatch :ref:`IPC <format-ipc>`
```
Comment (kou) [inline docs/source/java/vector_schema_root.rst:110]: It seems that we don't need this:

https://arrow.apache.org/docs/java/vector_schema_root.html#vectorschemaroot

> [ArrowRecordBatch](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.html) (a representation of a RecordBatch [IPC](https://arrow.apache.org/docs/format/Columnar.html#format-ipc) message).

Comment (Divyansh200102) [reply docs/source/java/vector_schema_root.rst:110]: So shall I remove this line no 110? 

> It seems that we don't need this:
> 
> https://arrow.apache.org/docs/java/vector_schema_root.html#vectorschemaroot
> 
> > [ArrowRecordBatch](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.html) (a representation of a RecordBatch [IPC](https://arrow.apache.org/docs/format/Columnar.html#format-ipc) message).

So shall I remove line no 110?

Comment (Divyansh200102) [reply docs/source/java/vector_schema_root.rst:110]: > It seems that we don't need this:
> 
> https://arrow.apache.org/docs/java/vector_schema_root.html#vectorschemaroot
> 
> > [ArrowRecordBatch](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.html) (a representation of a RecordBatch [IPC](https://arrow.apache.org/docs/format/Columnar.html#format-ipc) message).

So shall I remove line no 110?

Comment (Divyansh200102) [reply docs/source/java/vector_schema_root.rst]: Done @kou

**Comment by kou** [comment]
@github-actions crossbow submit preview-docs

**Comment by kou** [comment]
Hmm. The "IPC format" link markup is solved but the `** ... **` mark is broken...

http://crossbow.voltrondata.com/pr_docs/38276/format/Glossary.html#term-record-batch

> [**](http://crossbow.voltrondata.com/pr_docs/38276/format/Glossary.html#id1)In the [IPC format](http://crossbow.voltrondata.com/pr_docs/38276/format/Columnar.html#format-ipc) [**](http://crossbow.voltrondata.com/pr_docs/38276/format/Glossary.html#id3): the primitive unit of data.

**Comment by kou** [comment]
@AlenkaF Could you help us for reST markup?

File: docs/source/format/Glossary.rst [hunk 5]
```diff
@@ -174,7 +174,7 @@ Glossary
        .. seealso:: :term:`data type`
 
    record batch
-       **In the :ref:`IPC format <format-ipc>`**: the primitive unit
+       **In the :ref:`IPC format <format-ipc>` **: the primitive unit
```
Comment (AlenkaF) [inline docs/source/format/Glossary.rst:177]: It looks like a link in the bold text is not supported so I would do this:

```suggestion
      **In the** :ref:`IPC format <format-ipc>`: the primitive unit
```

File: docs/source/format/Glossary.rst [hunk 6]
```diff
@@ -174,7 +174,7 @@ Glossary
        .. seealso:: :term:`data type`
 
    record batch
-       **In the :ref:`IPC format <format-ipc>`**: the primitive unit
+      **In the** :ref:`IPC format <format-ipc>`: the primitive unit
```
Comment (AlenkaF) [inline docs/source/format/Glossary.rst:177]:
```suggestion
       **In the** :ref:`IPC format <format-ipc>`: the primitive unit
```

Sorry, one space is missing.

**Comment by AlenkaF** [comment]
@github-actions crossbow submit preview-docs

**Review by AlenkaF — APPROVED** [review]
Looks good now 👍 
http://crossbow.voltrondata.com/pr_docs/38276/format/Glossary.html#term-record-batch

**Pull request merged**